### PR TITLE
More precise locking for cache.Fetch

### PR DESCRIPTION
**Pull request by LisaLudique**
This PR is an attempt to make sharing of the cache resources more efficient amongst different go-routines.

### CODE REVIEW

File: internal/app/cache/cache.go [hunk 1]
```diff
@@ -85,8 +85,23 @@ func (c *cache) Fetch(key string) (*Response, error) {
...
+		value, found = c.cache.Get(key)
+		if !found {
+			// The entry was already evicted.
+			return nil, nil
+		}
+		resource, ok = value.(Resource)
+		if !ok {
+			return nil, fmt.Errorf("unable to cast cache value to type resource for key: %s", key)
+		}
+		if resource.isExpired(time.Now()) {
```
Comment (jessicayuen) [inline internal/app/cache/cache.go:99]: We're checking `isExpired` again in the case of a recent concurrent `set` right?

Comment (jessicayuen) [reply internal/app/cache/cache.go:99]: Would just document this a bit, but approach makes sense to me.

File: internal/app/cache/cache.go [hunk 2]
```diff
@@ -85,8 +85,23 @@ func (c *cache) Fetch(key string) (*Response, error) {
...
+			c.cache.Remove(key)
+			return nil, nil
+		}
+		// The entry should no longer be evicted.
+		return resource.resp, nil
```
Comment (jessicayuen) [inline internal/app/cache/cache.go:104]: Prob can delete this due to redundancy with the next line outside of the loop.

File: internal/app/cache/cache.go (hunk 1 above)
Comment (LisaLudique) [reply internal/app/cache/cache.go:102]: yep, added some comments

**Pull request merged**